### ISSUE TITLE
TypeError: Cannot read property '0' of undefined

### ISSUE THREAD

**Issue by artem1458** [Typescript-TDD/ts-auto-mock#238]
```
enum SOME_ENUM {
    FIRST = 'FIRST',
    SECOND = 'SECOND',
}

interface IBase {
    someField: string;
    anotherField: number;
}

interface IBugInterface extends Record<SOME_ENUM, IBase[]> {}
```
Can not extend Record, when first field in Record is ENUM, if it's string, everything's fine

**Comment by uittorio** [comment]
@artem1458 thanks for reporting this bug.

There is currently an issue extending enum.

I will look into it in the next few days. Is this blocking your development?

**Comment by artem1458** [comment]
> Is this blocking your development

It's not blocking issue, thanks for fast anser!

**Comment by uittorio** [comment]
I've created a couple of PR to prevent the compiler to fail.

One PR will just have a fail safe. This means that it will not mock the Mapped Type (Record) with the correct value but it will just be null. The compiler will not fail.

The second PR will fix the issue but its really bad implementation. Unfortunately I couldn't come up with a better solution because the scenario is really odd. 

I honestly will go for the first solution, fail safe. I am not sure how common is to extend Records with Enums

**Comment by uittorio** [comment]
I've released a new version of ts-auto-mock 1.5.6 where the compiler doesn't fail in this scenario but the type is not supported yet. 

https://typescript-tdd.github.io/ts-auto-mock/types-not-supported

**Comment by Pmyl** [comment]
can we close this one? It doesn't give error anymore and the fact that the type is unsupported is now in the documentation
